src/hotspot/share/opto/node.cpp

```diff
@@ -1472,16 +1467,8 @@ bool Node::rematerialize() const {
 bool Node::needs_anti_dependence_check() const {
   if( req() < 2 || (_flags & Flag_needs_anti_dependence_check) == 0 )
     return false;
-  else {
-#if INCLUDE_ZGC
-    if (UseZGC) {
-      if (ZBarrierSetC2::needs_anti_dependence_check(this)) {
-        return false;
-      }
-    }
-#endif
+  else
```

Comment:
the "else" is removed in origin patch

Reply:
The reason is to keep the change as little as possible.
Let's look into https://github.com/alibaba/dragonwell11/commit/3e9f7a74f7d0f6a13fc4f065fb88da8009f5abe8#diff-a9c13c0c2dc96fe9239116ac75460fdffbbbebd6d44540ccf7ec5811c8fbf79eL1470, the previous patch.
The change here actually undos the previous patch.
In the larger scope, if we want to publish **a single BIG patch** to change the experimental ZGC into production-ready one, we can squash all patches together. In this way, the BIG patch does not need to change any content in `bool Node::needs_anti_dependence_check()`. I believe it's better to keep the original version than to follow the upstream patch.